google/gax/api_callable.py

```diff
@@ -155,6 +156,107 @@ def inner(request):
     return inner
 
 
+def _construct_bundling(method_config, method_bundling_override):
+    """Helper for ``construct_settings()``."""
+    if ('bundle_options' in method_config and
+            'bundle_descriptor' in method_config):
+
+        if method_bundling_override == OPTION_INHERIT:
+            bundler = bundling.Executor(BundleOptions(
+                **method_config['bundle_options']))
+        else:
+            if method_bundling_override:
+                bundler = bundling.Executor(method_bundling_override)
+            else:
+                bundler = None
+
+        bundle_descriptor = BundleDescriptor(
+            **method_config['bundle_descriptor'])
+
+    else:
+        bundle_descriptor = None
+        bundler = None
+    return bundler, bundle_descriptor
+
+
+def _construct_retry(
+        method_config, method_retry_override, retry_codes_def, retry_params,
+        retry_names):
+    """Helper for ``construct_settings()``."""
+    if method_retry_override == OPTION_INHERIT:
+        codes = []
+        for code_set in retry_codes_def:
+            if code_set.get('name') == method_config['retry_codes_name']:
+                codes = [
+                    retry_names[name] for name in code_set['retry_codes']]
+                break
+
+        params_struct = None
+        for params in retry_params:
+            if params.get('name') == method_config['retry_params_name']:
+                params_struct = params.copy()
+        params_struct.pop('name')
+        for param in params_struct:
+            try:
+                params_struct[param] = int(params_struct[param])
+            except ValueError:
+                params_struct[param] = float(params_struct[param])
+
+        backoff_settings = BackoffSettings(**params_struct)
+
+        retry = RetryOptions(
+            retry_codes=codes, backoff_settings=backoff_settings)
+    else:
+        retry = method_retry_override
+
+    return retry
+
+
+def construct_settings(
+        configuration, bundling_override, retry_override, retry_names, timeout):
+    """Constructs a dictionary mapping method names to CallSettings.
+
```

Comment:
Added a yaml snippet.

The way it's set up currently, users do not override using a yaml; rather, they construct native Python object (e.g., BundleOptions, RetryOptions) and associate them with method names in a dictionary. The yaml is a static file that shouldn't need to be seen/edited by the user.